src/components/shapes/draw_newshape/defaults.js

```diff
@@ -16,6 +16,15 @@ module.exports = function supplyDrawNewShapeDefaults(layoutIn, layoutOut, coerce
         coerce('newshape.line.dash');
     }
 
+    var isLine = layoutIn.dragmode === 'drawline';
+    coerce('newshape.label.text');
+    coerce('newshape.label.textangle', isLine ? 'auto' : 0);
+    coerce('newshape.label.textposition', isLine ? 'middle' : 'middle center');
+    coerce('newshape.label.xanchor');
+    coerce('newshape.label.yanchor');
+    coerce('newshape.label.padding');
+    Lib.coerceFont(coerce, 'newshape.label.font', layoutOut.font);
```

Comment:
Similar to https://github.com/plotly/plotly.js/pull/6454/files#r1121794939, let's not coerce additional attributes when there is no label also set `dflt` of `textangle to *auto* for all shapes.
```js
    var text = coerce('newshape.label.text');
    if(text) {
      coerce('newshape.label.textangle');
      coerce('newshape.label.textposition', isLine ? 'middle' : 'middle center');
      coerce('newshape.label.xanchor');
      coerce('newshape.label.yanchor');
      coerce('newshape.label.padding');
      Lib.coerceFont(coerce, 'newshape.label.font', layoutOut.font);
    }
```

Reply:
> also set `dflt` of `textangle` to auto for all shapes

@archmoj Should this same change be made in `shapes/default.js` as well? Or are you saying the behaviour should be different between `shapes` and `draw_newshape`?

Reply:
oops just saw your other comment, all clear now!